proposals/features-contribute-lifecycle-scripts.md

```diff
@@ -0,0 +1,142 @@
+# [Proposal] Allow Dev Container Features to contribute lifecycle scripts
+
+Related to: https://github.com/devcontainers/spec/issues/60, https://github.com/devcontainers/spec/issues/181
+
+## Goal
+
+Allow Feature authors to provide lifecycle hooks for their Features during a dev container build.
+
+## Proposal
+
+ Introduce the following properties to [devcontainer-feature.json](https://containers.dev/implementors/features/#devcontainer-feature-json-properties), mirroring the behavior and syntax of the `devcontainer.json` lifecycle hooks.
+
+- "onCreateCommand"
```

Comment:
Mention that the value can be a string or an array of strings. The string variation is run with `/bin/sh -c ...` whereas the string array variation is executed directly. (This is also the case in the devcontainer.json.)

Reply:
Yes, good call out.

Reply:
Mentioned the semantics are the same as previously lifecycle scripts (with a link).